feat: return timestamp from `GetIDs`

tzdybal opened the pull request:
<!--
Please read and fill out this form before submitting your PR.

Please make sure you have reviewed our contributors guide before submitting your
first PR.

NOTE: PR titles should follow semantic commits: https://www.conventionalcommits.org/en/v1.0.0/
-->

## Overview

Enhanced the GetIDs functionality to include a timestamp in the response, both in the proto files and Go structures. Updated relevant files and added checks for the new field in the tests.

Resolves #101 
<!-- 
Please provide an explanation of the PR, including the appropriate context,
background, goal, and rationale. If there is an issue with this information,
please provide a tl;dr and link the issue. 

Ex: Closes #<issue number>
-->


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
	- Introduced a new `GetIDsResult` structure that includes both IDs and a timestamp, enhancing the context of retrieved data.
	- Added a `timestamp` field to the `GetIdsResponse` message for improved temporal context.
	- Implemented a new `check-proto-deps` target to streamline protobuf dependency management.
	- Updated plugin configuration to utilize `gocosmos` for better compatibility with the Cosmos ecosystem.
	- Added a new dependency on `buf.build/cosmos/gogo-proto` to enhance protobuf processing.

- **Bug Fixes**
	- Updated methods to ensure they return the new structured response, improving data handling and validation.

- **Chores**
	- Enhanced GitHub Actions workflows to set up the Go environment for executing commands.
	- Removed outdated scripts for protobuf generation, transitioning to a more efficient method.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

tzdybal committed:
feat: return timestamp from `GetIDs`

Enhanced the GetIDs functionality to include a timestamp in the response, both in the proto files and Go structures. Updated relevant files and added checks for the new field in the tests.

tzdybal committed:
chore: fix linter errors

gupadhyaya commented on proxy/jsonrpc/client.go:
```diff
@@ -31,6 +32,12 @@ type API struct {
 }
 
+// GetIDsResult is returned by `GetIDs` method.
+type GetIDsResult struct {
```
why is this re-declaration needed? there is already one in the da package?

tzdybal replied:
Good catch! I hoped that I will somehow hide this and use only for jsonrpc, but later I had to introduce it also in da.

tzdybal committed:
fix: remove unused struct

gupadhyaya commented:
could you fix golangci-lint? otherwise, LGTM

gupadhyaya commented on proto/da/da.proto:
```diff
@@ -79,6 +79,7 @@ message GetIdsRequest {
 // GetIdsResponse is the response type for the GetIds rpc method.
 message GetIdsResponse {
 	repeated ID ids = 1;
+  int64 timestamp = 2;
```
Did u look at google.protobuf.Timestamp instead of int64 for better precision and proto conversions?

tzdybal replied:
I did, but I got some issues with buf build setup.
I'll give it another try.

gupadhyaya replied:
Yes, i also had issue when doing go-sequencing. I used cosmos gogoproto. See https://github.com/rollkit/go-sequencing/blob/main/go.mod#L6

tzdybal replied:
Still can't get working CI....

tzdybal committed:
chore: Switch from Docker to 'plain' Buf for protobuf generation

Removed old protobuf generation shell scripts and updated Makefile to use Buf. Changed dependencies and updated configurations accordingly. This modernizes the setup and removes the dependency on Docker.

tzdybal committed:
refactor: use google.protobuf.Timestamp instead of int64

Replaced the int64 timestamp with google.protobuf.Timestamp in the GetIdsResponse message. Adjusted the generated code to handle the new timestamp type and updated the protobuf imports accordingly.

MSevey commented on .github/workflows/lint.yml:
```diff
@@ -52,5 +52,6 @@ jobs:
     timeout-minutes: 5
     steps:
       - uses: actions/checkout@v4
+      - run: echo "$GITHUB_WORKSPACE/go/bin" >> $GITHUB_PATH
```
is this leftover debug code? or does this job need `setup-go`?

MSevey commented on .github/workflows/proto.yml:
```diff
@@ -8,6 +8,7 @@ jobs:
     runs-on: ubuntu-latest
     timeout-minutes: 5
     steps:
+      - run: echo "$GITHUB_WORKSPACE/go/bin" >> $GITHUB_PATH
```
same, I think what the actual issue is that you need `setup-go` step

MSevey reviewed (changes requested):
I think the edits to the workflow files actually just need https://github.com/actions/setup-go

tzdybal committed:
ci: fix PATH issues

MSevey commented on .github/workflows/lint.yml:
```diff
@@ -52,5 +52,9 @@ jobs:
+      - name: set up go
+        uses: actions/setup-go@v5
+        with:
+          go-version: ${{ inputs.go-version }}
```
since we are editing these files anyways, might as well also make the improvement to use the go.mod file.

https://github.com/actions/setup-go?tab=readme-ov-file#getting-go-version-from-the-gomod-file

Then we can eliminate the `inputs.go-version` as well. This applies to all the use of go versions in the CI.

MSevey reviewed (changes requested):
One more improvement we might as well make since we are touching the CI files

tzdybal committed:
ci: use Go version from go.mod
